readme.md

```diff
@@ -0,0 +1,119 @@
+
+# How to install and use API
+
+#### Make sure you have the following requirements installed on your system before you start:
+- [Java 17](https://www.oracle.com/java/technologies/javase/jdk17-archive-downloads.html)
+- [Docker](https://docs.docker.com/get-docker/)
+- [MySQL database](https://www.mysql.com/downloads/)
+
+### Installation with docker
+
+Here are the steps to set up and run the Book Store API locally:
+
+1. **Clone the Repository:**
+
+      ```shell
+      git clone https://github.com/ArtemReizer/book-store.git
+      ```
+2. **Build with Maven:**
+   ```shell
+   mvn clean install
+   ```
+3. **Docker Image Creation:**
+   ```shell
+   docker build -t your-image-name .
+   ```
+4. **Docker Compose:**
+   ```shell
+   docker-compose build
+   docker-compose up
+   ```
+   
+### Installation without docker
+
+Here are the steps to set up and run the Book Store API locally without docker:
+
+1. **Clone the Repository:**
+      ```shell
+      git clone https://github.com/ArtemReizer/book-store.git
+      ```
+2. **Build with Maven:**
+   ```shell
+   mvn clean install
+   ```
+3. **Run application:**
+
+   Run application by using this command. Replace ***application*** in application.jar with actual name of the generated JAR file in target directory.
+   ```shell 
+      java -jar target/application.jar
+   ```
+4. **Configure a DB on your local machine:**
+
+   Create a *book_store* schema and update the username and password in *application.properties* with your credentials
+      **Congratulations, you've successfully set up and launched the Book Store API on your local machine!**
+
+## Postman Collection
+
+To simplify your experience, I've curated a Postman collection containing sample requests for different API endpoints. Feel free to download it [here](Book%20Store.postman_collection.json) and import it into your Postman workspace. This will enable you to swiftly initiate your testing and exploration of the Book Store API. Happy testing!
+
+## Challenges and Solutions
+
+During the development of the Online Book Store project, we encountered several challenges that shaped our journey in creating a robust e-commerce platform. Let's delve into these challenges and the solutions we implemented to overcome them.
+
+### Challenge 1: Data Modeling and Building Different Views
+
+**Issue**: Designing a flexible data model for books, users, orders, and categories while ensuring users only see necessary data.
+
+**Solution**: We utilized Spring Data JPA and Liquibase to create an adaptable schema. Additionally, we implemented a pattern Dto to present customized views of the data.
+
+### Challenge 2: Security
+
+**Issue**: Ensuring robust data security and user authentication to safeguard user information.
+
+**Solution**: We implemented Spring Security and JWT tokens, providing a secure environment and reliable user authentication mechanism.
```

Comment:
Please don`t use 'We', 'We' means that you were working on the project with someone else (fix in everywhere)